solutions/usaco-993.mdx

```diff
@@ -0,0 +1,84 @@
+---
+id: usaco-993
+source: USACO Gold 2020 January
+title: Time is Mooney
+author: Nathan Gong
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_time_gold_jan20.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(T_{\text{max}} \cdot (N + M))$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class TimeIsMooney {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("time");
+
+		// max time that we check (it can be any arbitrary large number as long 
+		// as it is sufficent for O(T_MAX * (N + M)) time)
+		final int T_MAX = 1001;
+		int n = io.nextInt();
+		int m = io.nextInt();
+		int c = io.nextInt();
+
+		// amount of moonies Bessie can make at each city
+		int[] values = new int[n];
+		for (int i = 0; i < values.length; i++) {
+			values[i] = io.nextInt();
+		}
+
+		List<List<Integer>> adj = new ArrayList<>();
+		for (int i = 0; i < n; i++) adj.add(new ArrayList<>());
+		for (int i = 0; i < m; i++) {
+			int n1 = io.nextInt() - 1;
+			int n2 = io.nextInt() - 1;
+			adj.get(n1).add(n2);
+		}
+
+		// dp[t][i] = max amount of moonies Bessie can make after t days, 
+		// ending at city i
```

Comment:
make this one line? it's ok if it extends by like 4 or 5 chars

Reply:
74 characters

Reply:
ok i'll just believe you